### PR TITLE
Tweaks for CaaSP

### PR DESCRIPTION
**Pull request by inercia**
Branch that contains all the adjustments needed in order to work on top of the CaaSP images.

For trying this, you can download the KVM image from the [CaaSP images repo](http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10/images/), copy it to the libvirt pool you use and run something _similar_ to this:

```
$ cd somewhere && git clone <this-repo-branch>
$ cd k8s-terraform && \
    ./k8s-setup -v \
    -F libvirt-caasp-local.profile \
    -V salt_dir=../k8s-salt \
    -V ssh_key=ssh/id_docker \
    -V volume_pool=personal \
    -V kube_minions_size=2 \
    -V cluster_prefix=caasp \
    -V net_cidr=10.17.15.0/24 \
    fmt
$ terraform apply
$ # now you should have the VMs ready
$ ssh -i ssh/id_docker  root@`terraform output ip_dashboard`
$ # in the dashboard machine
$ cd /tmp && sh ./provision-dashboard.sh --finish
```

## References

* Related to kubic-project/salt#3
* [CaaSP images](http://download.suse.de/ibs/SUSE:/SLE-12-SP2:/Update:/Products:/CASP10/images/)

## Note

In order to make this portable we will keep the Terraform copies to the CaaSP hosts, using some rw directories. The most important case is the Dashboard, where we are copying Salt scripts/pillars/etc that are already installed with rpms. So we are going to use a nasty technique: we will replace the _salt.manifest_ by our custom version, and this custom version will mount the rw directories in the Salt conatiners...

### CODE REVIEW

**Comment by mjura** [comment]
It looks great, but can you include info from #commnet 1 https://github.com/kubic-project/terraform/pull/1#issue-211134020 to README.md https://github.com/kubic-project/terraform/blob/master/README.md

File: libvirt-caasp-local.profile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Usage:
```
Comment (mssola) [inline libvirt-caasp-local.profile]: I don't understand why we're doing this. We don't do it for example for openSUSE in libvirt: we just add an extra profile that derives from the libvirt one and that's: there's no need to differentiate between `local` and `url`. I'd prefer simply passing the env. variable (or/and extending the `contrib/libvirt` script so this is abstracted away as we do with other setups).

File: provision/dashboard/manifests/salt.yaml [hunk 2]
```diff
@@ -51,10 +51,10 @@ spec:
         path: /tmp/salt/config/master.d
     - name: salt-master
       hostPath:
-        path: /srv/salt
+        path: /tmp/salt/sls
```
Comment (mssola) [inline provision/dashboard/manifests/salt.yaml]: I understand that `/srv` will no longer be available, but that's not the case for `/opt` iirc. Didn't we have other options for persisting stuff ? Or you really want this to go away after the first reboot ?

File: libvirt-caasp-local.profile (hunk 1 above)
Comment (inercia) [inline libvirt-caasp-local.profile]: Yeah, it makes sense to only have one profile...

File: provision/dashboard/manifests/salt.yaml (hunk 2 above)
Comment (inercia) [inline provision/dashboard/manifests/salt.yaml]: We don't care so much about persisting stuff between reboots. This Terraform environment is not close to what we will have in production: it is mostly for testing, so we do not care about rebooting...

Comment (mssola) [inline provision/dashboard/manifests/salt.yaml]: Fair enough

**Pull request merged**